dmaws/context.py

```diff
@@ -19,6 +20,18 @@ def __init__(self, home=os.getcwd(), dry_run=False):
         self.stacks = {}
         self.create_dependencies = False
 
+    def new_context(self, stage, environment, vars_files):
+        """Create a new context with different variables"""
+        ctx = Context()
+        ctx.stage = stage
+        ctx.environment = environment
+        ctx.load_variables(
+            files=ctx.get_variables_files(True, vars_files))
+        ctx.stacks = copy.deepcopy(self.stacks)
```

Comment:
With stacks replaced by 'built stacks' on the original context I'm not really sure whether we can be sure that this new context doesn't have any data carried over from the original environment. I think it might be better to do a fresh `ctx.load_stacks(stacks_file)`, unless I'm missing a reason why we need a copy.